### PR TITLE
Adding the ability to gather system and platform stats in AC.

### PR DESCRIPTION
**Pull request by menivaitsi**
Exposed via SOAP.

Trusty: https://ocd.appscale.com:8080/job/Daily%20Build/1180/

### CODE REVIEW

**Comment by menivaitsi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1181/

File: AppController/djinn.rb [hunk 1]
```diff
@@ -1787,13 +1787,13 @@ def job_start(secret)
...
-      # Print stats in the log recurrently; works as a heartbeat # mechanism.
+      # Print stats in the log recurrently; works as a heartbeat mechanism.
       if last_print < (Time.now.to_i - 60 * PRINT_STATS_MINUTES)
-        stats = get_stats(secret)
+        stats = JSON.parse(get_monitors(secret))
 
-        Djinn.log_info("--- Node at #{stats['ip']} is using" +
-          " #{stats['disk']}% disk, has #{stats['free_memory']}M memory" +
-          " available and knows about these apps #{stats['apps']}.")
+        Djinn.log_info("--- Node at #{stats['public_ip']} is using has " +
```
Comment (cdonati) [inline AppController/djinn.rb]: remove 'is using'

File: AppController/djinnServer.rb [hunk 2]
```diff
@@ -104,6 +104,8 @@ def on_init
     add_method(@djinn, "set_deployment_id", "secret")
     add_method(@djinn, "set_node_read_only", "read_only", "secret")
     add_method(@djinn, "set_read_only", "read_only", "secret")
+
```
Comment (obino) [inline AppController/djinnServer.rb]: Remove the empty line?

File: InfrastructureManager/infrastructure_manager_service.py [hunk 3]
```diff
@@ -129,7 +129,7 @@ def stop(self):
       logging.warn('Stop called on already stopped server')
 
 if __name__ == '__main__':
-  logging.basicConfig(level=logging.INFO,
+  logging.basicConfig(level=logging.DEBUG,
```
Comment (obino) [inline InfrastructureManager/infrastructure_manager_service.py]: Is this overwritten by the incoming parameters? if not, we may want to leave it at INFO level.

File: AppController/djinn.rb [hunk 4]
```diff
@@ -5656,4 +5675,54 @@ def stop_open()
...
   end
 
+  # Gathers App Controller and System Manager stats for this node.
+  #
+  # Args:
+  #   secret: The secret of this deployment.
+  # Returns:
+  #   A hash in string format containing system and platform stats for this
+  #     node.
+  def get_monitors(secret)
```
Comment (cdonati) [inline AppController/djinn.rb]: I think this name might be a little misleading since it returns info and not mechanisms for monitoring things.

Comment (cdonati) [inline AppController/djinn.rb]: maybe `get_combined_stats` or something

Comment (menivaitsi) [inline AppController/djinn.rb]: Can we settle for get_all_stats ?

File: InfrastructureManager/infrastructure_manager_service.py (hunk 3 above)
Comment (menivaitsi) [inline InfrastructureManager/infrastructure_manager_service.py]: I'll revert, it's leftover.

File: AppController/djinn.rb [hunk 5]
```diff
@@ -5656,4 +5675,54 @@ def stop_open()
...
+    all_stats = system_stats
+    all_stats["apps"] = []
+    controller_stats["apps"].each { |app_name, enabled|
+      if my_node.is_login?
+        # Get HAProxy requests.
+        Djinn.log_debug("Getting HAProxy stats for: #{app_name}")
+        total_reqs, reqs_enqueued, collection_time = get_haproxy_stats(app_name)
+        # Create the apps hash with useful information containing HAProxy stats.
+        all_stats["apps"] << {
+        app_name => {
```
Comment (cdonati) [inline AppController/djinn.rb]: Indent 2 more spaces.

File: AppController/djinn.rb [hunk 6]
```diff
@@ -5656,4 +5675,54 @@ def stop_open()
...
+    Djinn.log_debug("Controller stats: #{controller_stats}")
+
+    # Get stats from SystemManager.
+    imc = InfrastructureManagerClient.new(secret)
+    system_stats = imc.get_system_stats()
+    Djinn.log_debug("System stats: #{system_stats}")
+
+    # Combine all useful stats and return.
+    all_stats = system_stats
+    all_stats["apps"] = []
```
Comment (cdonati) [inline AppController/djinn.rb]: Does this need to be ordered? Could it just be a hash?

File: InfrastructureManager/infrastructure_manager_service.py (hunk 3 above)
Comment (cdonati) [inline InfrastructureManager/infrastructure_manager_service.py]: Does this need to be in debug for production?

File: InfrastructureManager/system_manager.py [hunk 7]
```diff
@@ -85,31 +109,46 @@ def get_memory_usage(self):
...
     """
+    if self.secret != secret:
+      return self.__generate_response(False,
+        InfrastructureManager.REASON_BAD_SECRET)
+
     monit_stats = subprocess.check_output(["monit", "summary"])
 
     monit_stats_dict = {}
     for line in monit_stats.split("\n"):
       tokens = " ".join(line.split()).split(" ")
```
Comment (cdonati) [inline InfrastructureManager/system_manager.py]: Can this just be `line.split()`? What is the extra join and split doing?

File: InfrastructureManager/system_manager.py [hunk 8]
```diff
@@ -120,3 +159,16 @@ def get_swap_usage(self):
...
+    """ Generate a system manager service response
+
+    Args:
+      status:  A boolean value indicating the status.
+      message: A str, the reason or failure.
+
+    Returns:
+      A dictionary containing the operation response.
+    """
+    logging.warn("Sending success = {0}, reason = {1}".format(status, message))
```
Comment (cdonati) [inline InfrastructureManager/system_manager.py]: I'm not sure if this should log a warning if it was a successful operation.

**Comment by menivaitsi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1182/

**Comment by menivaitsi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1183/

**Comment by menivaitsi** [comment]
Trusty: https://ocd.appscale.com:8080/job/Daily%20Build/1184/

**Comment by menivaitsi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1186/

**Pull request merged**